gooddata-sdk/tests/catalog/test_catalog_permission.py

```diff
@@ -164,3 +172,53 @@ def test_list_dashboard_permissions(test_config):
     assert len(dashboard_permissions.user_groups) == 0
     assert len(dashboard_permissions.users) == 0
     assert len(dashboard_permissions.rules) == 0
+
+
+@gd_vcr.use_cassette(str(_fixtures_dir / "put_declarative_organization_permissions.yaml"))
+def test_put_and_get_declarative_organization_permissions(test_config):
+    expected_json_path = _current_dir / "expected" / "declarative_organization_permissions.json"
+    sdk = GoodDataSdk.create(host_=test_config["host"], token_=test_config["token"])
+    catalog_declarative_permissions_initial = sdk.catalog_permission.get_declarative_organization_permissions()
+
+    assert len(catalog_declarative_permissions_initial) == 1
+    _assert_organization_permissions_id(catalog_declarative_permissions_initial)
+    assert set(org_permission.name for org_permission in catalog_declarative_permissions_initial) == {"MANAGE"}
+
+    with open(expected_json_path, "r", encoding="utf-8") as f:
+        data = json.load(f)
+
+    declarative_organization_permissions = []
+    for permission in data:
+        declarative_organization_permissions.append(CatalogDeclarativeOrganizationPermission.from_api(permission))
+
+    sdk.catalog_permission.put_declarative_organization_permissions(declarative_organization_permissions)
+
+    catalog_declarative_permissions_after_put = sdk.catalog_permission.get_declarative_organization_permissions()
+    assert len(catalog_declarative_permissions_after_put) == 2
+    _assert_organization_permissions_id(catalog_declarative_permissions_after_put)
+    assert set(org_permission.name for org_permission in catalog_declarative_permissions_after_put) == {
+        "MANAGE",
+        "SELF_CREATE_TOKEN",
+    }
+
+
+@gd_vcr.use_cassette(str(_fixtures_dir / "manage_organization_permissions.yaml"))
+def test_manage_organization_permissions(test_config):
+    sdk = GoodDataSdk.create(host_=test_config["host"], token_=test_config["token"])
+
+    # assign permissions to adminGroup
+    sdk.catalog_permission.manage_organization_permissions(
+        [
+            CatalogOrganizationPermissionAssignment(
+                assignee_identifier=CatalogAssigneeIdentifier(id="adminGroup", type="userGroup"),
+                permissions=["MANAGE", "SELF_CREATE_TOKEN"],
+            )
+        ],
+    )
+
+    catalog_declarative_permissions_initial = sdk.catalog_permission.get_declarative_organization_permissions()
+    assert len(catalog_declarative_permissions_initial) == 2
+    assert set(org_permission.name for org_permission in catalog_declarative_permissions_initial) == {
+        "MANAGE",
+        "SELF_CREATE_TOKEN",
+    }
```

Comment:
I am still determining the tests. There is no AfterAll or AfterEach mechanism as is, for example, known in Java. Therefore, if you put something, then it stays put. There should be a `try-finally` block where the change is reverted in the final block back to the original state.